HttpClient.go

```diff
@@ -0,0 +1,58 @@
+package main
+
+import (
+	"errors"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/go-kit/kit/metrics"
+)
+
+type httpClient interface {
+	Do(*http.Request) (*http.Response, error)
+}
+
+// DoerFunc implements HTTPClient
+type DoerFunc func(*http.Request) (*http.Response, error)
```

Comment:
```suggestion
type doerFunc func(*http.Request) (*http.Response, error)
```

I don't think we need to export this type.